components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -77,6 +100,25 @@ func (r *NotebookReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Log.WithValues("notebook", req.NamespacedName)
 
+	event := &v1.Event{}
+	var getEventErr error
+	getEventErr = r.Get(ctx, req.NamespacedName, event)
+	if getEventErr == nil {
+		involvedNotebook := &v1beta1.Notebook{}
+		involvedNotebookKey := types.NamespacedName{Name: nbNameFromEvent(event), Namespace: req.Namespace}
+		if err := r.Get(ctx, involvedNotebookKey, involvedNotebook); err != nil {
+			log.Error(err, "unable to fetch Notebook by looking at event")
+			return ctrl.Result{}, ignoreNotFound(err)
+		}
+		r.eventRecorder.Eventf(involvedNotebook, event.Type, event.Reason,
+			"Reissued from %s/%s: %s", strings.ToLower(event.InvolvedObject.Kind), event.InvolvedObject.Name, event.Message)
+		return ctrl.Result{}, nil
```

Comment:
if no error, we should continue the reconcile and not return right?

Reply:
I guess it wouldn't harm. But since the state is reconciled after reissuing the event, I thought the rest of the reconcile method would be basically a no-op. Anyway, you're right, in this way we are also consistent with the rest of the code.